File: question_one/travelItineraries.js

const attractions = [
    { id: 1, name: " Nairobi GamePark", category: "Park", crowdLevel: "low" },
    { id: 2, name: "Nairobi Museam", category: "Art", crowdLevel: "medium" },
    { id: 3, name: "Bomas of Kenya", category: "Historical", crowdLevel: "high" }
];

const events = [
    { date: "2025-08-12", name: "Art Exhibition", location: "Nairobi Museum" },
    { date: "2025-02-22", name: "Concert", location: "Uhuru Park" }
];

const weatherForecast = {
    "2025-05-21": { condition: "sunny", temperature: 70 },
    "2025-05-22": { condition: "rainy", temperature: 48 }
};

function fetchAttractions(userPreferences) {
    return attractions.filter(attraction => 
        userPreferences.interests.includes(attraction.category)
    );
}
function fetchLocalEvents(date) {
    return events.filter(event => event.date === date);
}
function fetchWeather(date) {
    return weatherForecast[date];
}
function generateItinerary(userPreferences, travelDates) {
    let itinerary = [];
    travelDates.forEach(date => {
        let dailyPlan = {
            date: date,
            activities: []
        };
        const availableAttractions = fetchAttractions(userPreferences);
        const localEvents = fetchLocalEvents(date);
        const weather = fetchWeather(date);
        if (localEvents.length > 0) {
            dailyPlan.activities.push(...localEvents);
        }
        availableAttractions.forEach(attraction => {
            if (isAttractionAvailable(attraction, weather)) {
                dailyPlan.activities.push(attraction);
            }
        });
        dailyPlan.route = optimizeRoute(dailyPlan.activities);
        itinerary.push(dailyPlan);
    });

    return itinerary;
}
function isAttractionAvailable(attraction, weather) {
    return (attraction.crowdLevel !== "high" || weather.condition !== "rainy");
}
function optimizeRoute(activities) {
    let activities = ["dance","biking","skiing"]
    return activities; 
}
const userPreferences = {
    interests: ["Art", "Nature"],
    budget: 100
};

// const travelDates = ["2025-05-21", "2025-05-22"];
// const itinerary = generateItinerary(userPreferences, travelDates);
// console.log(itinerary);

File: question_one/travelItineraries.py
Attractions = [
    {"id": 1, "name": "Museum of Art", "category": "Art", "crowdLevel": "low"},
    {"id": 2, "name": "City Park", "category": "Nature", "crowdLevel": "medium"},
    {"id": 3, "name": "Historical Site", "category": "History", "crowdLevel": "high"}
]

events = [
    {"date": "2025-05-21", "name": "Art Exhibition", "location": "Museum of Art"},
    {"date": "2025-05-22", "name": "Outdoor Concert", "location": "City Park"}
]

weather_forecast = {
    "2025-05-21": {"condition": "sunny", "temperature": 75},
    "2025-05-22": {"condition": "rainy", "temperature": 68}
}

def attractions(user_preferences):
    for attraction in Attractions:
        if attraction["category"] in user_preferences["interests"]:
            print (attraction)
attractions(user_preferences = {
    "interests": ["Art", "Nature"],
    "budget": 100
})
def local_events(date):
    for event in events :
        if event["date"] == date:
            return [event]
def weather(date):
    return weather_forecast[date]

def is_attraction_available(attraction, weather):
        if attraction["crowdLevel"] == "high" and weather["condition"] == "rainy":
            return false

def itinerary(user_preferences, travel_dates):
    itinerary = []

    for date in travel_dates:
        daily_plan = {"date": date, "activities": []}
        available_attractions = attractions(user_preferences)
        local_events = local_events(date)
        weather = weather(date)

        if local_events:
            daily_plan["activities"].extend(local_events)

        for attraction in available_attractions:
            if is_attraction_available(attraction, weather):
                daily_plan["activities"].append(attraction)

        daily_plan["route"] = optimize_route(daily_plan["activities"])
        itinerary.append(daily_plan)

    return itinerary

def route_optimization(activities):
    return activities 